src/main/java/com/rapid7/client/dcerpc/mslsad/objects/LSAPR_POLICY_AUDIT_EVENTS_INFO.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2017, Rapid7, Inc.
+ *
+ * License: BSD-3-clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * * Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ */
+package com.rapid7.client.dcerpc.mslsad.objects;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+import com.rapid7.client.dcerpc.io.PacketInput;
+import com.rapid7.client.dcerpc.io.ndr.Alignment;
+import com.rapid7.client.dcerpc.io.ndr.Unmarshallable;
+
+/**
+ * The LSAPR_POLICY_AUDIT_EVENTS_INFO structure is used to set and query the system's auditing rules. The
+ * LsaQueryInformationPolicy and LsaSetInformationPolicy functions use this structure when their InformationClass
+ * parameters are set to PolicyAuditEventsInformation.
+ *
+ * <pre>
+ *      typedef struct _LSAPR_POLICY_AUDIT_EVENTS_INFO {
+ *          unsigned char AuditingMode;
+ *          [size_is(MaximumAuditEventCount)] unsigned long* EventAuditingOptions;
+ *          [range(0,1000)] unsigned long MaximumAuditEventCount;
+ *      } LSAPR_POLICY_AUDIT_EVENTS_INFO,
+ *       *PLSAPR_POLICY_AUDIT_EVENTS_INFO;
+ * </pre>
+ *
+ * @see <a href =
+ * "https://msdn.microsoft.com/en-us/library/windows/desktop/ms721901(v=vs.85).aspx">POLICY_AUDIT_EVENTS_INFO
+ * structure</a>
+ */
+public class LSAPR_POLICY_AUDIT_EVENTS_INFO implements Unmarshallable {
```

Comment:
Thoughts on making  POLICY_INFORMATION interface that extends Unmarshallable so it's more apparent that they are all types of policy information?

Reply:
I agree, to follow spec we should have `POLICY_INFORMATION` as well, which knows to read its tag as a short, validate it, and then read the expected `Unmarshallable`. However, I ran into another layer of generics, and it started to look ugly when using it with `Response`. I'll take another pass at it though.